portability-transfer/src/main/java/org/datatransferproject/transfer/WorkerModule.java

```diff
@@ -121,7 +126,13 @@ AppCredentialStore getBucketStore() {
   Exporter getExporter(ImmutableList<TransferExtension> transferExtensions) {
     TransferExtension extension =
         findTransferExtension(transferExtensions, JobMetadata.getExportService());
-    extension.initialize(context);
+    DelegatingExtensionContext serviceSpecificContext = new DelegatingExtensionContext(context);
+    serviceSpecificContext.registerOverrideService(
+        MetricRecorder.class,
+        new ServiceAwareMetricRecorder(
+            extension.getServiceId(),
+            context.getService(DtpInternalMetricRecorder.class)));
```

Comment:
For clear error checking, do we need to do a precondition that DtpInternalMetricRecorder exists ?

Reply:
Good point, instead of adding it here I added it in ServiceAwareMeticRecorder so it covers both cases.